### PR TITLE
nixos/ydotool: Make group configurable

### LINKED ISSUE #317013: `programs.ydotool.group` property
This would allow users to configure `users.users.<name>.extraGroups` and the like by referring to this setting rather than indirect references like `config.users.groups.ydotool.name` or magic strings like `"ydotool"`.

### PR DESCRIPTION
**Pull request by l0b0**
## Description of changes

Allows users to refer to `config.programs.ydotool.group` rather than
hard-coding "ydotool".

Allows users to override the group name for whatever reason.

This closes #317013.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/programs/ydotool.nix
Comment (OPNA2608) [inline nixos/modules/programs/ydotool.nix:17]: https://github.com/NixOS/nixpkgs/blob/78be25a21623159fac10384f2c245e7161b49fe4/lib/options.nix#L98-L103

```suggestion
      ydotoold system service and {command}`ydotool` for members of
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Also would you mind keeping the reformating of the tests as a separate commit from any changes to it? Makes it abit harder to grok.

**Comment by OPNA2608** [comment]
CC @quantenzitrone since we init'd the module & test (maybe we should be maintainers of the module as well?).

File: nixos/tests/ydotool.nix
Comment (quantenzitrone) [reply nixos/tests/ydotool.nix:170]: i think so
tested in the repl, currently legacyPackages.x86_64-linux.nixosTests.ydotool.wayland.meta.maintainers is empty

**Comment by quantenzitrone** [comment]
i am module maintainer, did you mean package maintainers?

**Review by quantenzitrone — COMMENTED** [review]
other than the `meta.maintainers` being on the wrong level for the `nixosTests`, this lgtm

File: nixos/tests/ydotool.nix
Comment (l0b0) [reply nixos/tests/ydotool.nix]: Fixed, thanks!

**Comment by l0b0** [comment]
> Also would you mind keeping the reformating of the tests as a separate commit from any changes to it? Makes it abit harder to grok.

I did the refactoring in the first commit. Did you mean a separate PR?

**Comment by quantenzitrone** [comment]
@ofborg test ydotool.headless ydotool.wayland ydotool.x11 ydotool.customGroup

**Comment by quantenzitrone** [comment]
i can't get the tests to build on my machine, the error message is very cryptic

**Comment by quantenzitrone** [comment]
veery weird
```
Rewriting #!/nix/store/7hnr99nxrd2aw6lghybqdmkckq60j6l9-python3-3.11.9/bin/python3.11 to #!/nix/store/7hnr99nxrd2aw6lghybqdmkckq60j6l9-python3-3.11.9
wrapping `/nix/store/r0dpr6j195ik8m5salfyzwiaanam6dlq-nixos-test-driver-1.1/bin/nixos-test-driver'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Running phase: installCheckPhase
## run mypy
test_driver/driver.py:109: error: No return value expected  [return-value]
Found 1 error in 1 file (checked 8 source files)
```

i tried switching to the normal nix implementation (i normally use lix), but it didn't change anything.

```sh
$ nix --version
nix (Nix) 2.18.2
```

I can't approve a pr I can't build, but i won't block it either.
I may try a few more things to get it to build, but I'm currently out of ideas.

**Comment by l0b0** [comment]
Which exact command are you running?

**Comment by quantenzitrone** [comment]
I'm using flakes and nix-command, so I use `nix build .#nixosTests.ydotool.headless`.
But `nix-build -A nixosTests.ydotool` (what does this even try to build? `nixosTests.ydotool` is not a derivation) and `nix-build -A nixosTests.ydotool.headless` fail with the same error.

**Comment by l0b0** [comment]
On commit 8a0e50ac8b292f8eff8849aba4c2e2d1ca5f90b6, `nix-build -A nixosTests.ydotool` and `nix build .#nixosTests.ydotool.headless` both pass. I've tested the first one on two machines and the last one on one machine.

**Comment by quantenzitrone** [comment]
Wait, the force push from bd84309e7380 to 8a0e50ac8b29 actually fixed it. :woozy_face::tada:

**Comment by l0b0** [comment]
@OPNA2608 @quantenzitrone How's this?

**Review by OPNA2608 — APPROVED** [review]
LGTM. Tests pass locally, doesn't break the Lomiri test.

File: nixos/modules/programs/ydotool.nix
Comment (SuperSandro2000) [inline nixos/modules/programs/ydotool.nix:32]:
```suggestion
  config = lib.mkIf cfg.enable {
    users.groups."${config.programs.ydotool.group}" = { };
```
that's a bit over engineered. We can just hardcode this

File: nixos/modules/programs/ydotool.nix
Comment (SuperSandro2000) [inline nixos/modules/programs/ydotool.nix:40]:
```suggestion
        RuntimeDirectory = "ydotoold";
```

File: nixos/modules/programs/ydotool.nix
Comment (SuperSandro2000) [inline nixos/modules/programs/ydotool.nix:88]:
```suggestion
      YDOTOOL_SOCKET = "/run/ydotoold/socket";
```

File: nixos/modules/programs/ydotool.nix
Comment (l0b0) [reply nixos/modules/programs/ydotool.nix:32]: I disagree. It's used in three places, and it's not obvious how they're related without this variable.

**Review by quantenzitrone — APPROVED** [review]

**Pull request merged**
